src/device-state.coffee

```diff
@@ -406,6 +410,37 @@ module.exports = class DeviceState extends EventEmitter
 			apps = _.keyBy(_.map(appsArray, singleToMulticontainerApp), 'appId')
 			return { apps, config: deviceConf }
 
+	restoreBackup: (targetState) =>
+		@setTarget(targetState)
+		.then =>
+			appId = _.keys(targetState.local.apps)[0]
+			if !appId?
+				throw new Error('No appId in target state')
+			volumes = targetState.local.apps[appId].volumes
+			backupPath = path.join(constants.rootMountPoint, 'mnt/data/backup')
+			rimraf(backupPath) # We clear this path in case it exists from an incomplete run of this function
```

Comment:
Given we're already using shell commands further down, and that we're already heavily tied to linux, is it worth bringing in this extra dependency for this task?

Reply:
It was already used by some of our dependencies, but true that we don't really needed here... I changed it to another execAsync (see [line 48](https://github.com/balena-io/balena-supervisor/pull/835/files#diff-0011437c96d80cf058a252cafc68dc0bR48))